test/widgets/new_dm_sheet_test.dart

```diff
@@ -0,0 +1,257 @@
+        check(find.text('Charlie Carter')).findsNothing();
+        check(find.text('Bob Brown')).findsNothing();
+      });
+
+      testWidgets('search is case-insensitive', (tester) async {
+        await setupSheet(tester, users: testUsers);
+        await tester.enterText(find.byType(TextField), 'alice');
+        await tester.pump();
+        check(find.text('Alice Anderson')).findsOne();
+
+        await tester.enterText(find.byType(TextField), 'ALICE');
+        await tester.pump();
+        check(find.text('Alice Anderson')).findsOne();
+      });
+
+      testWidgets('partial name and last name search handling', (tester) async {
+        await setupSheet(tester, users: testUsers);
+
+        await tester.enterText(find.byType(TextField), 'Ali');
+        await tester.pump();
+        check(find.text('Alice Anderson')).findsOne();
+        check(find.text('Bob Brown')).findsNothing();
+        check(find.text('Charlie Carter')).findsNothing();
+
+        await tester.enterText(find.byType(TextField), 'Anderson');
+        await tester.pump();
+        check(find.text('Alice Anderson')).findsOne();
+        check(find.text('Charlie Carter')).findsNothing();
+        check(find.text('Bob Brown')).findsNothing();
+
+        await tester.enterText(find.byType(TextField), 'son');
+        await tester.pump();
+        check(find.text('Alice Anderson')).findsOne();
+        check(find.text('Charlie Carter')).findsNothing();
+        check(find.text('Bob Brown')).findsNothing();
+      });
+
+      testWidgets('shows empty state when no users match', (tester) async {
+        await setupSheet(tester, users: testUsers);
+        await tester.enterText(find.byType(TextField), 'Zebra');
+        await tester.pump();
+        check(find.text('No users found')).findsOne();
+        check(find.text('Alice Anderson')).findsNothing();
+        check(find.text('Bob Brown')).findsNothing();
+        check(find.text('Charlie Carter')).findsNothing();
+      });
+    });
+
+    group('user selection', () {
+      testWidgets('selecting a user', (tester) async {
+        final user = eg.user(fullName: 'Test User');
+        await setupSheet(tester, users: [user]);
+        Finder findNextButton() => find.widgetWithText(GestureDetector, 'Next');
+
+        check(find.byIcon(Icons.circle_outlined)).findsOne();
+        check(find.byIcon(Icons.check_circle)).findsNothing();
+
+        var nextButton = tester.widget<GestureDetector>(findNextButton());
+        check(nextButton.onTap).isNull();
+        final userTile = find.ancestor(
+          of: find.text(user.fullName),
+          matching: find.byType(InkWell));
+        await tester.tap(userTile);
+        await tester.pump();
+        check(find.byIcon(Icons.check_circle)).findsOne();
+        check(find.byIcon(Icons.circle_outlined)).findsNothing();
+        nextButton = tester.widget<GestureDetector>(findNextButton());
+        check(nextButton.onTap).isNotNull();
+      });
```

Comment:
nit:

```suggestion
      testWidgets('selecting a user', (tester) async {
        final user = eg.user(fullName: 'Test User');
        final userTileFinder = find.ancestor(
          of: find.text(user.fullName),
          matching: find.byType(InkWell));
        await setupSheet(tester, users: [user]);

        final nextButton = tester.widget<GestureDetector>(
          find.widgetWithText(GestureDetector, 'Next'));
        check(find.byIcon(Icons.circle_outlined)).findsOne();
        check(find.byIcon(Icons.check_circle)).findsNothing();
        check(nextButton.onTap).isNull();

        await tester.tap(userTileFinder);
        await tester.pump();
        check(find.byIcon(Icons.circle_outlined)).findsNothing();
        check(find.byIcon(Icons.check_circle)).findsOne();
        check(nextButton.onTap).isNotNull();
      });
```

By breaking the checks into two stanzas, we can make this a bit more readable. (We don't need to call `tester.widget()` twice to see updates to `nextButton.onTap`.)

Reply:
> We don't need to call `tester.widget()` twice to see updates to `nextButton.onTap`.

I tried implementing this change, but I'm running into some issues - the tests are failing when I remove the second call.
When I only get the button once before the tap and then check its `onTap` property after the UI update, the test fails as if the property isn't being updated.
Could you help me understand the correct approach here?

Reply:
I see. Yeah, it looks like the return values of the `tester.widget` call before and after the pump are different. Let's keep calling them again then.